c++/CMakeLists.txt

```diff
@@ -51,6 +46,10 @@ if(CAPNP_LITE AND BUILD_TOOLS)
   message(WARNING "Command-line tools will not be built with CAPNP_LITE.")
 endif()
 
+if(BUILD_TOOLS AND EXTERNAL_CAPNP)
+  message(SEND_ERROR "EXTERNAL_CAPNP cannot be set with BUILD_TOOLS.")
```

Comment:
It's convenient, but very fragile. Changes in one project can break the other in weird ways (e.g. compiler flags in your project that don't work with Cap'n Proto), and global targets get conflated (e.g. `make install` will always install Cap'n Proto tools and headers, even if you only wanted the library). It's not feasible to protect against all possibilities.